src/swap/v2/index.ts

```diff
@@ -333,22 +401,188 @@ function getFixedOutputSwapQuote({
+      "Output amount exceeds available liquidity."
+    );
+  }
+
+  if (isSwapAssetInAmountLow(Number(swapInputAmount))) {
+    throw new SwapQuoteError(
+      SwapQuoteErrorType.LowSwapAmountError,
+      "Swap amount is too low."
+    );
   }
 
   return {
-    assetInID,
-    assetInAmount: swapInputAmount,
-    assetOutID: assetOut.id,
-    assetOutAmount,
-    swapFee: Number(totalFeeAmount),
-    rate:
-      convertFromBaseUnits(decimals.assetOut, Number(assetOutAmount)) /
-      convertFromBaseUnits(decimals.assetIn, Number(swapInputAmount)),
-    priceImpact
+    type: SwapQuoteType.Direct,
+    data: {
+      pool,
+      quote: {
+        assetInID,
+        assetInAmount: swapInputAmount,
+        assetOutID,
+        assetOutAmount,
+        swapFee: Number(totalFeeAmount),
+        rate:
+          convertFromBaseUnits(decimals.assetOut, Number(assetOutAmount)) /
+          convertFromBaseUnits(decimals.assetIn, Number(swapInputAmount)),
+        priceImpact
+      }
+    }
   };
 }
 
+/**
+ * @returns A quote for a fixed input swap. Does NOT execute any transactions.
+ */
+async function getFixedInputSwapQuote({
+  amount,
+  assetInID,
+  assetOutID,
+  decimals,
+  isSwapRouterEnabled,
+  network,
+  pool
+}: {
+  amount: number | bigint;
+  assetInID: number;
+  assetOutID: number;
+  decimals: {assetIn: number; assetOut: number};
+  network: SupportedNetwork;
+  pool: V2PoolInfo | null;
+  isSwapRouterEnabled?: boolean;
+}): Promise<SwapQuote> {
+  const quotePromises: Promise<SwapQuote>[] = [];
+
+  if (pool) {
+    quotePromises.push(
+      new Promise((resolve, reject) => {
+        try {
+          const quote = getFixedInputDirectSwapQuote({
+            amount,
+            assetInID,
+            assetOutID,
+            decimals,
+            pool
+          });
+
+          resolve({
+            type: SwapQuoteType.Direct,
+            data: {
+              pool,
+              quote
+            }
+          });
+        } catch (error) {
+          reject(error);
+        }
+      })
+    );
+  } else {
+    quotePromises.push(
+      Promise.reject(
+        new SwapQuoteError(
+          SwapQuoteErrorType.NoAvailablePoolError,
+          "There is not an available pool for this swap. However, you can enable swap router and try to perform this swap with multiple pools."
+        )
+      )
+    );
+  }
+
+  if (isSwapRouterEnabled) {
+    quotePromises.push(
+      (async () => {
+        const data = await getSwapRoute({
+          amount,
+          assetInID,
+          assetOutID,
+          swapType: SwapType.FixedInput,
+          network
+        });
+
+        if (data.route.length < 2) {
+          throw new SwapQuoteError(
+            SwapQuoteErrorType.SwapRouterNoRouteError,
+            "Swap router couldn't find a route for this swap."
+          );
+        }
+
+        return {
+          type: SwapQuoteType.Router,
+          data
+        };
+      })()
```

Comment:
```suggestion
async function generateSwapRouterQuote() {
        const data = await getSwapRoute({
          amount,
          assetInID,
          assetOutID,
          swapType: SwapType.FixedInput,
          network
        });

        if (data.route.length < 2) {
          throw new SwapQuoteError(
            SwapQuoteErrorType.SwapRouterNoRouteError,
            "Swap router couldn't find a route for this swap."
          );
        }

        return {
          type: SwapQuoteType.Router,
          data
        };
      }
      
    quotePromises.push(generateSwapRouterQuote())      
```

But actually, we should be able to handle `data.route.length < 2` case inside getSwapRoute directly, so we can do

```suggestion
    quotePromises.push(getSwapRoute({
          amount,
          assetInID,
          assetOutID,
          swapType: SwapType.FixedInput,
          network
        }))
```